packages/teleport-shared/src/node-handlers/node-to-jsx/index.ts

```diff
@@ -88,7 +88,13 @@ const generateElementNode = (
 ) => {
   const { dependencies, nodesLookup } = params
   const { elementType, children, key, attrs, dependency, events } = node.content
-  const elementTag = children ? createJSXTag(elementType) : createSelfClosingJSXTag(elementType)
+  // Append app for only local dependencies
+  const elementTag =
+    dependency && dependency.type === 'local' && options.customElementTag
+      ? createJSXTag(options.customElementTag(elementType))
```

Comment:
maybe here you can compute the name of the tag before and then leave the same logic as above (previous line 91) where the tag is created either as a selfclosing or regular JSX tag